test/mongodb.test.js

```diff
@@ -1406,6 +1406,68 @@ describe('mongodb connector', function() {
     });
   });
 
+  it('findOrCreate should properly support field projection (on create) - object', function() {
+    const query = {where: {title: 'Kopria post'}, fields: {comments: 1}};
```

Comment:
I think we should run these new tests for all connectors supporting `findOrCreate` to ensure consisten behavior across databases.

I am ok to keep those tests here for short term, to get this fix landed soon.

As the next step, I would like these tests to be moved to juggler's "manipulation" test suite, see 
[test/manipulation.test.js#L1557-L1622](https://github.com/strongloop/loopback-datasource-juggler/blob/fcac276955a58c4aabcd2b602af046f45cfc73fe/test/manipulation.test.js#L1557-L1622).

Reply:
I agree to keep this change for the future